tutorials/migration.md

```diff
@@ -4,96 +4,846 @@ layout: intro
+you may want to set. In particular for this tutorial we recommend setting:
+
+- `org.eclipse.smarthome.persistence:default=mapdb` - default persistence, use
+whatever was set in `persistence:default` in openhab.cfg
+- `autoapprove:enabled=false` - makes it so you must manually approve 
+automatically discovered Things in the Inbox (see below)
+- `org.eclipse.smarthome.links:autoLinks=false` - prevents openHAB from 
+automatically creating an Item for each automatically discovered Thing.
+
+Next open for editing `addons.cfg`. This file contains parameters related to 
+add-ons including allowing one to choose which add-ons to install. For now just
+set the following parameters:
+
+- `package = minimal` - minimal or standard are good choices here. Standard
+will include the standard UIs, though you can choose those individually later
+- `remote = true` - required for the installation of legacy 1.x and 
+experimental add-ons
+- `legacy = true` - required to install the 1.9 version of an add-on for which 
+there is a 2.0 native version
+- `experimental = true` - required to install experimental add-ons
+- `binding = ` - a comma separated list of the bindings from openHAB 1.x that
+you use. If you are in doubt of the name of a binding, look in openhab.cfg for
+that binding's configurations. The first part of the tag in openhab.cfg will be
+that name of the binding. For example, the configuration parameters for the 
+MQTT Binding start with "mqtt" in openhab.cfg so the name of the binding is 
+"mqtt". For those bindings that have a 2.0 version, use the name followed by a 
+"1", for example, to use the 1.9 SNAPSHOT version of the Astro binding instead
+of the 2.0 version, use the name "astro1". You can find the list of 2.0 
+bindings [here](../addons/bindings.html).
+- `ui = ` - if you intend on using PaperUI include "paper", if you use zwave I 
+recommend "habmin". The list of UIs are [here](../addons/uis.html).
+- `action = ` - the list of action add-ons you used in openHAB 1.x.
+- `transformation = ` = the list of transformations you use. Unlike in openHAB 1,
+one must install transformations separately.
+- `voice = ` - see [here](../addons/voice.html)
+- `misc = ` - myopenhab, homekit, etc. Do not list myopenhab at this time, 
+instructions for it are below.
+
+If you have any doubt as to what add-ons are available and in which category 
+they belong you can follow the instructions below for listing the bundles in 
+the Karaf console or install PaperUI and browse the list through it.
+
+This is a great opportunity to identify those add-ons you are not actively 
+using and avoid their installation.
+
+When you save this file openHAB will begin the download (if using the Online 
+version) and installing the add-ons.
+
+Note: if running in Docker, create <openHAB 2 home>/userdata/persistence prior to saving the 
+file if there are any persistence add-ons.
+
+An example addons.cfg:
+
+~~~~
```

Comment:
There where two times code fences did not get compiled for me. 
1. when there are spaces behind them
2. Discourse: At some point Discourse stops rendering. I do not recall what the reason was. Something with special characters or Headlines... Anyhow that's not related to Jekyll,  just put it in here.